specifyweb/frontend/js_src/lib/components/SkeletonLoaders/AttachmentGallery.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import ContentLoader from 'react-content-loader';
+
+import { userPreferences } from '../Preferences/userPreferences';
+
+const DEFAULT_FETCH_LIMIT = 20;
+
+export function AttachmentGallerySkeleton(): JSX.Element {
+  const [motionPref] = userPreferences.use('general', 'ui', 'reduceMotion');
+
+  function createRectangles() {
+    const rectWidth = 22;
+    const rectHeight = 22;
+    const rectRadius = 2;
+    let rectX = 5; // Starting x position
+    let rectY = 0; // Starting y position
+    const rectangles = []; // Array to hold the generated rectangles
+
+    for (let i = 0; i < DEFAULT_FETCH_LIMIT; i++) {
```

Comment:
there are just 3 `for (let ...` loops in the entire sp7 codebase. we generally try to avoid mutation whenever possible

Could replace you code with something like this:
```ts
Array.from({length:DEFAULT_FETCH_LIMIT}, (_,index)=>
  <rect
    ...
    // example logic, not sure if it matches what you had:
    x={rectWidth * index + 5}
    y={rectHeight * index}
    // you can also use something like `(index % 4)` or `Math.ceil(index / 4)` if needed
  />
)
```

Reply:
Benefits of not mutating:
- When debugging you can execute the code in your console, without fearing side-effects messing up things
- Each iteration of the loop does not depend on the previous one, thus way less stuff to keep in the head when trying to understand what it is doing
- Matches the general react philosophy of not mutating data
- More more and more benefits (look up online if you are interested)

Reply:
kind of related, but I don't like how this library requires you to hardcode all these pixel values.

i.e, something like this would have been so much simpler:
```tsx
<div className="flex flex-wrap gap-4">
  {Array.from({length: DEFAULT_FETCH_LIMIT}, ()=>
    <div className="skeleton-square w-64 h-64 rounded"  />
  )}
</div>
```

(flexbox takes care of all the positioning, rather than dealing with ugly pixel values and doing math)

Isn't that simpler and less error prone and easier to implement?